src/protocols/UBX.hpp

```diff
@@ -0,0 +1,135 @@
+#ifndef GPS_UBLOX_PROTOCOLS_UBX_HPP
+#define GPS_UBLOX_PROTOCOLS_UBX_HPP
+
+#include <stdint.h>
+#include <stddef.h>
+
+#include <array>
+#include <vector>
+
+#include <gps_ublox/protocols/Parser.hpp>
+
+namespace gps_ublox
+{
+    namespace protocols {
+        class UBX : public Parser
```

Comment:
Now that I had to code a little bit in the driver, I'm puzzled as to the purpose of this. The only protocol we'll ever need in this driver is the UBX protocol. The other protocols are meant to interface with other systems, UBX is both the most powerful (since it's the only one that can be used to configure the unit) and a superset of the others.

I have the impression that removing the class level - possibly if you'd like by renaming `protocols` into `ubx` or just `protocol` would make a lot of sense, and simplify the code quite a bit.

Reply:
You are probably right about RTCM, but I am not sure about NMEA. I can't say for sure UBX provides everything NMEA provides. For instance, I can't see standard deviations in UBX (GPGBS and GPGST in NMEA).

In any case, this was an early design choice based on my initial understanding of the documentation that I will be glad to revert if we can. Would you please take a look at GPGBS and GPGST NMEA message (and possibly others) and confirm we don't need those?

Reply:
I'm positive we do not need NMEA.

PVT contains most (maybe even all) of the information we need for the navigation solution, other messages will be needed for internal status monitoring and for satellite constellation information.

The bottom line is that Ublox basically design the UBX protocol to cover all the functionality of their chips (most vendors do that, going as far as having proprietary extensions to NMEA). If an information is not present in UBX, I think we can safely assume it won't be set in NMEA (one can set blank fields in NMEA).

In any case, even if NMEA was needed, there are good NMEA parsing libraries out there.

Reply:
Great.. I will remove the abstraction layer then